runtime/native/jvm.cpp

```diff
@@ -765,7 +865,7 @@ JVM_GetClassDeclaredFields(JNIEnv *env, jclass ofClass, jboolean publicOnly) {
                 signature = staticFields[sidx].sig;
                 slot = -(i+1); // 0 ambiguity
             }
-            jstring nameString = env->NewStringUTF(name);
+            jstring nameString = internJString(env->NewStringUTF(name));
```

Comment:
Is there a reason we're interning field names? (e.g. does OpenJDK or Oracle do this by default?)

Reply:
Hmm, not sure, but I think this was necessary to get the atomic long stuff to work because it only calls intern on the string argument and not on the name of the field, which makes me suspect this is what they do. I can check up on this though.

Reply:
Maybe this is the reason (from String.intern()):

> All  literal strings and string-valued constant expressions are interned. String literals are defined in section 3.10.5 of the The Java™ Language Specification.

I have to check, but perhaps that includes something like all field,class and method names